src/task_impl/atomic_task.rs

```diff
@@ -7,7 +7,25 @@ use core::cell::UnsafeCell;
 use core::sync::atomic::AtomicUsize;
 use core::sync::atomic::Ordering::{Acquire, Release};
 
-/// A coordinated `Task` handle enabling concurrent operations on a task.
+/// A synchronization primitive for task notification.
+///
+/// `AtomicTask` will coordinate concurrent notifications with the consumer
+/// potentially "updating" the underlying task to notify. This is useful in
+/// scenarios where a computation completes in another thread and wants to
+/// notify the consumer, but the consumer is in the process of being migrated to
+/// a new logical task.
+///
+/// Roughly, consumers should call `park` before checking the result of a
```

Comment:
That block is handling concurrent calls to `park`. Those states are hit when a concurrent thread is currently in the park path, in which case the op is aborted as we consider the other thread has "won" the race.